python/seldon_core/utils.py

```diff
@@ -27,19 +35,14 @@
 logger = logging.getLogger(__name__)
 
 
-ENV_MODEL_NAME = "PREDICTIVE_UNIT_ID"
-ENV_MODEL_IMAGE = "PREDICTIVE_UNIT_IMAGE"
-NONIMPLEMENTED_MSG = "NOT_IMPLEMENTED"
-
-model_name = os.environ.get(ENV_MODEL_NAME, f"{NONIMPLEMENTED_MSG}")
-image_name = os.environ.get(
-    ENV_MODEL_IMAGE, f"{NONIMPLEMENTED_MSG}:{NONIMPLEMENTED_MSG}"
-)
-
-
 def get_request_path():
+    model_name = get_model_name()
     if model_name == NONIMPLEMENTED_MSG:
         return {}
+    image_name = get_image_name()
+    assert (
```

Comment:
Why the extra assertion? Just falling back to non-implemented seemed to work fine.

Reply:
I felt that if `ENV_MODEL_NAME` env variable is set, it should reference a valid docker image. So `ENV_MODEL_IMAGE` needs to be set as well? Perhaps to run locally this is not the case though but I am not sure.

I guess the question is really whether it is odd to return `{"proper_model_name": "non-implemented"}`? Happy to take the extra assert if it is fine for users to set one env variable and not the other.

Reply:
Hmm.. I guess it should be okay. In k8s both should be set by operator so won't matter much. As long as tests will pas let's leave it as it is.